### PR TITLE
Openapi 3

### PR DESCRIPTION
**Pull request by butonic**
I was confused by the examples for `/share` and converted the spec from swagger to openapi which allows using multiple examples. IMO we should update to openapi to better document how the api should be used.

This PR consists of four commits:
1. [convert to openapi v3](https://github.com/cs3org/OCM-API/commit/f51de0da29e8e6cf5bfb68d07b8508d1c233a6e7) (it explicitly uses `application/hal+json` for a lot of responses, but I don't know how critical that is)
2. [use multiple share examples](https://github.com/cs3org/OCM-API/commit/73a88b112be51d255c10bf5477881683560aae20)
3.  [use multiple notification examples](https://github.com/cs3org/OCM-API/commit/4e21c01e15451f185f2251e1a008a7f57b5e8a76)
4. [deprecate protocol options property](https://github.com/cs3org/OCM-API/commit/0d8a7bb536d97a52eb1d4a327fc60478a87b89c3)

Let me know what you think.

### CODE REVIEW

**Comment by butonic** [comment]
urgh, now I see https://github.com/cs3org/OCM-API/pull/64

**Comment by glpatcern** [comment]
> urgh, now I see #64

I think we can close that one that is now outdated, and concentrate on this one.

**Comment by glpatcern** [comment]
@butonic could you please rebase this? I think it's very valuable at this stage - and with no outstanding change/tuning of the specs - to move to OpenAPI. And we could then envisage to tag OCM v1.2.0.

**Comment by mickenordin** [comment]
Current version of spec can easily be converted to openapi 3 using the following command:

```
npx -p swagger2openapi swagger2openapi https://raw.githubusercontent.com/cs3org/OCM-API/develop/spec.yaml --outfile spec.yaml -t 3.0.3
```

**Comment by michielbdejong** [comment]
I ran that and pushed it to the `openapi3` branch. Does https://cs3org.github.io/OCM-API/docs.html?branch=openapi3&repo=OCM-API&user=cs3org#/paths/~1invite-accepted/post look OK? If so we can switch to it!

**Review by glpatcern — CHANGES_REQUESTED** [review]
> I ran that and pushed it to the `openapi3` branch. Does https://cs3org.github.io/OCM-API/docs.html?branch=openapi3&repo=OCM-API&user=cs3org#/paths/~1invite-accepted/post look OK? If so we can switch to it!

Well, again we have quite a number of ongoing PRs, and this branch is already in conflict with the current HEAD. I'd postpone this until we settle them (and before tagging 1.2).

**Review by michielbdejong — APPROVED** [review]

**Comment by michielbdejong** [comment]
I ran the script on latest `develop` and force-pushed.
How can we test if it is OK?

**Comment by michielbdejong** [comment]
Ah, https://cs3org.github.io/OCM-API/docs.html?branch=openapi-3&repo=OCM-API&user=cs3org looks OK I think?

**Comment by mickenordin** [comment]
> Ah, https://cs3org.github.io/OCM-API/docs.html?branch=openapi-3&repo=OCM-API&user=cs3org looks OK I think?

Yes! Looks good to me!

**Review by mickenordin — APPROVED** [review]
It is very hard to follow all the changes in detail, but looking at the spec viewer I don't see any apparent errors at least.

**Review by glpatcern — APPROVED** [review]
I've just removed the empty lines, otherwise as Micke said it's hard to follow the changes but as long as the rendering looks correct let's go!
